do_cmake.sh

```diff
@@ -72,18 +72,24 @@ elif type ccache > /dev/null 2>&1 ; then
     ARGS+=" -DWITH_CCACHE=ON"
 fi
 
-cxx_compiler="g++"
-c_compiler="gcc"
-# 20 is used for more future-proof
-for i in $(seq 20 -1 11); do
-  if type -t gcc-$i > /dev/null; then
-    cxx_compiler="g++-$i"
-    c_compiler="gcc-$i"
-    break
-  fi
-done
-ARGS+=" -DCMAKE_CXX_COMPILER=$cxx_compiler"
-ARGS+=" -DCMAKE_C_COMPILER=$c_compiler"
+
+if [ -n "$DCMAKE_C_COMPILER" ] && [ -n "$DCMAKE_CXX_COMPILER" ]; then
```

Comment:
neither `DCMAKE_C_COMPILER` nor `DCMAKE_CXX_COMPILER` is an env variable. i am not sure if this works as you expect. probably you could parse / check `$@` instead? see also https://github.com/ceph/ceph/blob/54197bf38ca72027864edc977f19cfbc0d1bf4ec/do_cmake.sh#L107

Reply:
Ah, that's right:
```
    $DRY_RUN ./do_cmake.sh $cmake_opts $@ || return 1
```

Thanks, updating!